src/docfx/build/toc/TocMap.cs

```diff
@@ -254,6 +276,42 @@ void RemoveInvalidServicePage()
         }
     }
 
+    private (HashSet<FilePath> originalReferenceTOCs, List<FilePath> targetReferenceTOCs)
+        GetOriginalReferenceTocWithTargetReferenceToc(IEnumerable<FilePath> tocFilesFromBuildScope)
+    {
+        var originalReferenceTOCs = new HashSet<FilePath>();
+        var originalReferenceTocFilePathMap = new Dictionary<string, FilePath>(StringComparer.OrdinalIgnoreCase);
+        var targetReferenceTOCs = new List<FilePath>();
+
+        foreach (var joinTOCConfig in _config.JoinTOC)
+        {
+            if (!string.IsNullOrEmpty(joinTOCConfig.OriginalReferenceToc))
+            {
+                var relativePathForOriRefToc = Path.GetRelativePath(_buildOptions.DocsetPath, joinTOCConfig.OriginalReferenceToc!);
+                if (!originalReferenceTocFilePathMap.ContainsKey(joinTOCConfig.OriginalReferenceToc))
```

Comment:
Why do we have to loop over `tocFilesFromBuildScope` to get the TOC file path for original reference TOC? Could we just calculate by `Path.Combine(_buildOptions.DocsetPath, joinTOCConfig.OriginalReferenceToc)`?

Reply:
`FilePath` contains a lot other properties and it difficult to recovery all of them here. So I want to loop to find the exact one